### PR TITLE
feat(DIA-342): wire up partnerCollectorProfile

### PR DESCRIPTION
**Pull request by araujobarret**
[DIA-342]

- Wires up PartnerCollectorProfile type to the respective model(presenter) in Gravity
- deprecates fromProfile in favor of partnerCollectorProfile in a conversation
- adds firstNameLastInitial to the CollectorProfileType

[DIA-342]: https://artsyproduct.atlassian.net/browse/DIA-342?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/schema/v2/partner/__tests__/partnerCollectorProfile.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    collectorProfile: {
+      type: CollectorProfileType,
+      resolve: async (
+        { partnerId, collectorId },
+        _args,
+        { partnerCollectorProfileLoader }
+      ) => {
+        if (!partnerCollectorProfileLoader) return
+
+        const data = await partnerCollectorProfileLoader({
```
Comment (egdbear) [inline src/schema/v2/partner/__tests__/partnerCollectorProfile.ts:20]: **suggestion:** This shouldn't be part of the `__tests__` directory? 
Also could we make this a bit more defensive? For instance, adding a `try catch` block.

**Review by egdbear — CHANGES_REQUESTED** [review]

File: src/schema/v2/partner/partnerCollectorProfile.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            partnerId,
+            userId: collectorId,
+          })
+
+          return {
+            ...data.collector_profile,
+            follows_profile: data.follows_profile,
+          }
+        } catch (error) {
+          console.error("PartnerCollectorProfileType", error)
```
Comment (egdbear) [inline src/schema/v2/partner/partnerCollectorProfile.ts:31]: **nitpick** 

```suggestion
          console.error("[schema/v2/partner/partnerCollectorProfile] Error:", error)
```

File: .vscode/settings.json [hunk 3]
```diff
@@ -43,6 +43,6 @@
   "typescript.surveys.enabled": false,
   "editor.defaultFormatter": "esbenp.prettier-vscode",
   "editor.codeActionsOnSave": {
-    "source.fixAll.eslint": true
+    "source.fixAll.eslint": "explicit"
```
Comment (egdbear) [inline .vscode/settings.json:46]: **question:** is this something related to the eslint version in metaphysics?

**Review by egdbear — APPROVED** [review]
Looks good to me!

File: .vscode/settings.json (hunk 3 above)
Comment (araujobarret) [reply .vscode/settings.json]: it is already "explicit" probably my branch was outdated, after a rebase that was solved 👍

**Comment by araujobarret** [comment]
@mzikherman I'm leaving today for some ooo but @egdbear can help with some changes if necessary!

File: src/schema/v2/conversation/index.ts [hunk 4]
```diff
@@ -303,10 +304,22 @@ export const ConversationType = new GraphQLObjectType<any, ResolverContext>({
...
+          partnerId: to_id,
+          collectorId: from_id,
+        }
+      },
+    },
     fromProfile: {
       description:
         "The collector profile of the user who initiated the conversation",
       type: CollectorProfileType,
+      deprecationReason: "Use `partnerCollectorProfile` instead",
```
Comment (egdbear) [inline src/schema/v2/conversation/index.ts:322]: **Question:** We used to have:

`fromUser` - deprecated. "It will be inaccessible to partners in future versions. Prefer using `fromProfile` instead."

`fromProfile` - it is becoming deprecated in favor of `partnerCollectorProfile`.

Should we address the previous deprecations?

File: _schemaV2.graphql [hunk 5]
```diff
@@ -14392,6 +14398,10 @@ enum PartnerClassification {
   PRIVATE_DEALER
 }
 
+type PartnerCollectorProfile {
```
Comment (mzikherman) [inline _schemaV2.graphql:14401]: This type doesn't seem to be doing anything currently, but I'm thinking that other metadata about the collector (like the `follows_profile` thing, some Exchange-backed summary data, and so on), will be added here.

In that case, cool! 👍 

However, `PartnerCollectorProfile` feels like a weird name for something that includes `collectorProfile: CollectorProfile` as a field. Maybe this type should be called `CollectorResume` or something of that sort? (should double-check the name). I think that would go a long way to help clearing up initial confusion as to why there are two types that have `-CollectorProfile` suffix.

File: src/schema/v2/partner/partnerCollectorProfile.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+        try {
+          const data = await partnerCollectorProfileLoader({
+            partnerId,
+            userId: collectorId,
+          })
+
+          return {
+            ...data.collector_profile,
+            follows_profile: data.follows_profile,
```
Comment (mzikherman) [inline src/schema/v2/partner/partnerCollectorProfile.ts:28]: NIce! Not seeing where `follows_profile` (can be camel-cased here) is being used within this type?

**Review by mzikherman — COMMENTED** [review]
Somehow a new type called `PartnerCollectorProfile` which includes `CollectorProfile` initially seems odd, although I think this is the right abstraction, but we shouldn't call it `PartnerCollectorProfile`.

File: _schemaV2.graphql (hunk 5 above)
Comment (egdbear) [reply _schemaV2.graphql:14401]: I wonder how we can transform data from [this presenter](https://github.com/artsy/gravity/blob/main/app/presenters/partner_collector_profile_presenter.rb#L1) to this type to avoid a double call to the resolver

**Comment by egdbear** [comment]
As per @mzikherman suggestion I replaced PartnerCollectorProfile with the CollectorResume.

```type CollectorResume {
  collectorProfile: CollectorProfileType

  # The Collector follows the Gallery profile
  hasPartnerFollow: Boolean
}
```

File: src/schema/v2/conversation/index.ts [hunk 7]
```diff
@@ -303,10 +304,41 @@ export const ConversationType = new GraphQLObjectType<any, ResolverContext>({
...
+
+        try {
+          const data = await partnerCollectorProfileLoader({
+            partnerId: to_id,
+            userId: from_id,
+          })
+
+          return {
+            collector_profile: data.collector_profile,
+            follows_profile: data.follows_profile,
```
Comment (egdbear) [inline src/schema/v2/conversation/index.ts:326]: `followsProfile` vs `hasPartnerFollow` - we might want to consolidate this naming and avoid confusion downstream

File: src/schema/v2/conversation/collectorResume.ts [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+
+export const CollectorResume = new GraphQLObjectType<any, ResolverContext>({
+  name: "CollectorResume",
+  fields: () => ({
+    collectorProfile: {
+      type: CollectorProfileType,
+      resolve: ({ collector_profile }) => collector_profile,
+    },
+    hasPartnerFollow: {
+      type: GraphQLBoolean,
```
Comment (dzucconi) [inline src/schema/v2/conversation/collectorResume.ts:13]: Should be non-nullable

Comment (dzucconi) [reply src/schema/v2/conversation/collectorResume.ts:13]: We try to use `is` prefixing naming for booleans, if possible/makes sense.

File: src/schema/v2/conversation/collectorResume.ts [hunk 9]
```diff
@@ -0,0 +1,18 @@
+import { GraphQLObjectType, GraphQLBoolean } from "graphql"
+import { CollectorProfileType } from "schema/v2/CollectorProfile/collectorProfile"
+import { ResolverContext } from "types/graphql"
+
+export const CollectorResume = new GraphQLObjectType<any, ResolverContext>({
+  name: "CollectorResume",
+  fields: () => ({
+    collectorProfile: {
+      type: CollectorProfileType,
+      resolve: ({ collector_profile }) => collector_profile,
```
Comment (dzucconi) [inline src/schema/v2/conversation/collectorResume.ts:10]: Should this be non-nullable as well?

File: src/schema/v2/conversation/collectorResume.ts (hunk 8 above)
Comment (egdbear) [reply src/schema/v2/conversation/collectorResume.ts:13]: how about `isCollectorFollowingPartner`?

File: src/schema/v2/conversation/collectorResume.ts (hunk 9 above)
Comment (egdbear) [reply src/schema/v2/conversation/collectorResume.ts:10]: yes!

File: src/schema/v2/conversation/index.ts [hunk 10]
```diff
@@ -303,10 +304,41 @@ export const ConversationType = new GraphQLObjectType<any, ResolverContext>({
...
+          )
+          return null
+        }
+      },
+    },
     fromProfile: {
       description:
         "The collector profile of the user who initiated the conversation",
       type: CollectorProfileType,
+      deprecationReason: "Use `partnerCollectorProfile` instead",
```
Comment (mzikherman) [inline src/schema/v2/conversation/index.ts:341]: #minor reason still mentions `partnerCollectorProfile` naming

**Review by mzikherman — DISMISSED** [review]
